### PR TITLE
feat: updates with probablity and severity info

### LINKED ISSUE #790: [Feat]: update safety ratings with examples of probability and severity
### Is your feature request related to a problem? Please describe.

_No response_

### Describe the solution you'd like

include examples of how to use probability and severity in safety filters

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ghchinoy**
# Description

Thank you for opening a Pull Request!
Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [X] Follow the [`CONTRIBUTING` Guide](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/CONTRIBUTING.md).
- [X] You are listed as the author in your notebook or README file.
  - [X] Your account is listed in [`CODEOWNERS`](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/.github/CODEOWNERS) for the file(s).
- [X] Make your Pull Request title in the <https://www.conventionalcommits.org/> specification.
- [ ] Ensure the tests and linter pass (Run `nox -s format` from the repository root to format).
- [ ] Appropriate docs were updated (if necessary)

Fixes #790 🦕

### CODE REVIEW

**Comment by ghchinoy** [comment]
FYI "SPII" is an abbreviation for "Sensitive Personally Identifiable Information" which is in our docs and copied in directly.

File: gemini/responsible-ai/gemini_safety_ratings.ipynb [hunk 1]
```diff
@@ -1,865 +1,1138 @@
...
+      },
+      "source": [
+        "## Understanding Blocked Responses\n",
+        "\n",
+        "The documentation for [FinishReason](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/GenerateContentResponse#finishreason) contains some more detailed explanations.\n",
+        "\n",
+        "For example, the previous response was blocked with the `finish_reason: SAFETY`, indicating that\n",
+        "> The token generation was stopped as the response was flagged for safety reasons. NOTE: When streaming the Candidate.content will be empty if content filters blocked the output.\n",
+        "\n",
+        "As of this writing, the table from the [FinishReason docs](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/GenerateContentResponse#finishreason) have been reproduced below, but please look at the docs for the definitive explanations\n"
```
Comment (holtskinner) [inline gemini/responsible-ai/gemini_safety_ratings.ipynb:1083]: Not sure the link needs to be added here again.

File: gemini/responsible-ai/gemini_safety_ratings.ipynb [hunk 2]
```diff
@@ -1,865 +1,1138 @@
...
+      "metadata": {
+        "id": "l2v6VnECf-fC"
+      },
+      "source": [
+        "## Understanding Blocked Responses\n",
+        "\n",
+        "The documentation for [FinishReason](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/GenerateContentResponse#finishreason) contains some more detailed explanations.\n",
+        "\n",
+        "For example, the previous response was blocked with the `finish_reason: SAFETY`, indicating that\n",
+        "> The token generation was stopped as the response was flagged for safety reasons. NOTE: When streaming the Candidate.content will be empty if content filters blocked the output.\n",
```
Comment (holtskinner) [inline gemini/responsible-ai/gemini_safety_ratings.ipynb:1081]:
```suggestion
        "> The token generation was stopped as the response was flagged for safety reasons. NOTE: When streaming the `Candidate.content` will be empty if content filters blocked the output.\n",
```

File: gemini/responsible-ai/gemini_safety_ratings.ipynb [hunk 3]
```diff
@@ -1,865 +1,1138 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "l2v6VnECf-fC"
+      },
+      "source": [
+        "## Understanding Blocked Responses\n",
+        "\n",
+        "The documentation for [FinishReason](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/GenerateContentResponse#finishreason) contains some more detailed explanations.\n",
```
Comment (holtskinner) [inline gemini/responsible-ai/gemini_safety_ratings.ipynb:1078]:
```suggestion
        "The documentation for [`FinishReason`](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/GenerateContentResponse#finishreason) contains some more detailed explanations.\n",
```

File: gemini/responsible-ai/gemini_safety_ratings.ipynb [hunk 4]
```diff
@@ -1,865 +1,1138 @@
...
+        "--- | ---\n",
+        "FINISH_REASON_UNSPECIFIED\t| The finish reason is unspecified.\n",
+        "STOP\t| Natural stop point of the model or provided stop sequence.\n",
+        "MAX_TOKENS\t| The maximum number of tokens as specified in the request was reached.\n",
+        "SAFETY |\tThe token generation was stopped as the response was flagged for safety reasons. NOTE: When streaming the Candidate.content will be empty if content filters blocked the output.\n",
+        "RECITATION\t| The token generation was stopped as the response was flagged for unauthorized citations.\n",
+        "OTHER\tAll | other reasons that stopped the token generation\n",
+        "BLOCKLIST |\tThe token generation was stopped as the response was flagged for the terms which are included from the terminology blocklist.\n",
+        "PROHIBITED_CONTENT\t| The token generation was stopped as the response was flagged for the prohibited contents.\n",
+        "SPII\t| The token generation was stopped as the response was flagged for Sensitive Personally Identifiable Information (SPII) contents."
```
Comment (holtskinner) [inline gemini/responsible-ai/gemini_safety_ratings.ipynb:1103]:
```suggestion
        "`FINISH_REASON_UNSPECIFIED`\t| The finish reason is unspecified.\n",
        "`STOP`\t| Natural stop point of the model or provided stop sequence.\n",
        "`MAX_TOKENS`\t| The maximum number of tokens as specified in the request was reached.\n",
        "`SAFETY` |\tThe token generation was stopped as the response was flagged for safety reasons. NOTE: When streaming the `Candidate.content` will be empty if content filters blocked the output.\n",
        "`RECITATION`\t| The token generation was stopped as the response was flagged for unauthorized citations.\n",
        "`OTHER`\tAll | other reasons that stopped the token generation\n",
        "`BLOCKLIST` |\tThe token generation was stopped as the response was flagged for the terms which are included from the terminology blocklist.\n",
        "`PROHIBITED_CONTENT`\t| The token generation was stopped as the response was flagged for the prohibited contents.\n",
        "`SPII`\t| The token generation was stopped as the response was flagged for Sensitive Personally Identifiable Information (SPII) contents."
```

**Review by holtskinner — APPROVED** [review]

File: gemini/responsible-ai/gemini_safety_ratings.ipynb (hunk 1 above)
Comment (ghchinoy) [reply gemini/responsible-ai/gemini_safety_ratings.ipynb]: good catch, removed 2nd reference to doc in block

**Pull request merged**